lib/events/api.go

```diff
@@ -614,6 +614,11 @@ type IAuditLog interface {
 	// WaitForDelivery waits for resources to be released and outstanding requests to
 	// complete after calling Close method
 	WaitForDelivery(context.Context) error
+
+	// StreamSessionEvents streams all events from a given session recording. An error is returned on the first
+	// channel if one is encountered. Otherwise it is simply closed when the stream ends.
+	// The event channel is not closed on error to prevent race conditions in downstream select statements.
+	StreamSessionEvents(ctx context.Context, sessionID session.ID, startIndex int) (chan apievents.AuditEvent, chan error)
```

Comment:
Bit of a nit, but IMO it would be preferable if the channel were either always closed or never close.  I'm of the opinion that inconsistent channel closure is hard to reason about, and makes it easy to make subtle (hard to detect in reviews) mistakes in both producer and consumer logic.

One potential fix that would require minimal changes would be to simply never close the event channel, and always close the error channel (i.e. `defer close(e)`).  Very difficult to misuse since the closure signal is always "I got a value from the error channel" and checking if an error is nil is a very well established pattern.  In many cases, checking wouldn't event be necessary in the consuming context.  Ex:

```golang
for {
    select {
    case event := <-eventC:
        // ...
    case err := <-errC:
        return trace.Wrap(err)
    }
}
```

Reply:
Hmm, that's a neat idea and allows for neat consumer logic but IMO it goes away from what I see as the "this makes sense" behaviour of message stream -> message end marked by channel close OR error is signaled. This is highly subjective and I could go either way. Do we have any consistent standard for this across the codebase or documented anywhere? (I can't find anything)